types/next-auth/next-auth-tests.ts

```diff
@@ -5,15 +5,43 @@
  * in the sense of typing and call signature consistency. They
  * are not intended as functional tests.
  */
-import Providers from 'next-auth/providers';
-import Adapters, { Adapter, EmailSessionProvider, Profile, Session, VerificationRequest } from 'next-auth/adapters';
+import Providers, { AppProvider, DefaultProviders } from 'next-auth/providers';
+import Adapters, { Adapter, EmailAppProvider, Profile, Session, VerificationRequest } from 'next-auth/adapters';
 import * as client from 'next-auth/client';
-import * as JWT from 'next-auth/jwt';
+import * as JWTType from 'next-auth/jwt';
 import NextAuth, * as NextAuthTypes from 'next-auth';
-import { GenericObject, SessionBase, NextApiRequest, NextApiResponse } from 'next-auth/_utils';
 import { IncomingMessage, ServerResponse } from 'http';
 import { Socket } from 'net';
 
+/* copied from next@10.0 */
+export interface NextApiRequest extends IncomingMessage {
+    query: {
+        [key: string]: string | string[];
+    };
+    cookies: {
+        [key: string]: string;
+    };
+    body: any;
+    env: any;
+    preview?: boolean;
+    previewData?: any;
+}
+
+export type Send<T> = (body: T) => void;
+
+export type NextApiResponse<T = any> = ServerResponse & {
+    send: Send<T>;
+    json: Send<T>;
+    status: (statusCode: number) => NextApiResponse<T>;
+    redirect(url: string): NextApiResponse<T>;
+    redirect(status: number, url: string): NextApiResponse<T>;
+    setPreviewData: (data: object | string, options?: {
+        maxAge?: number;
+    }) => NextApiResponse<T>;
+    clearPreviewData: () => NextApiResponse<T>;
+};
+/* ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */
```

Comment:
We could use here the types from [`./_next.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b7fafcf9fc2ea8401fd0ee0e5637e0b73d0f51a1/types/next-auth/_next.d.ts) or is there any specific reason to keep them duplicate? 🧐

Reply:
@euxn23 noting that I would like to have this answered/addressed before merging 🙏🏽

Reply:
@lluia 

Reason of this is relative import is forbidden by dtslint 😢 

>  Test file should not use a relative import. Use a global import as if this were a user of the package. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-relative-import-in-test.md